src/Apps/CollectorProfile/Routes/Saves2/Components/SavesArtworks.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  ArtworkFilterContextProvider,
+  Counts,
+} from "Components/ArtworkFilter/ArtworkFilterContext"
+import { SortOptions } from "Components/SortFilter"
+import { FC } from "react"
+import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
+import { useRouter } from "System/Router/useRouter"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { Flex, Join, MoreIcon, Spacer, Text } from "@artsy/palette"
+import { updateUrl } from "Components/ArtworkFilter/Utils/urlBuilder"
+import { SavesArtworksGridPlaceholder } from "Apps/CollectorProfile/Routes/Saves2/Components/SavesPlaceholders"
+
+interface SavesArtworksQueryRendererProps {
+  collectionID: string
+  initialPage?: number
+  initialSort?: string
+}
+
+interface SavesArtworksProps extends SavesArtworksQueryRendererProps {
+  collection: SavesArtworks_collection$data
+  collectionID: string
+  relay: RelayRefetchProp
+}
+
+const sortOptions: SortOptions = [
+  { value: "SAVED_AT_DESC", text: "Recently Saved" },
+  { value: "SAVED_AT_ASC", text: "Oldest First" },
+]
+const defaultSort = sortOptions[0].value
+
+const SavesArtworks: FC<SavesArtworksProps> = ({
+  collection,
+  collectionID,
+  relay,
+}) => {
+  const { match } = useRouter()
+
+  const counts: Counts = {
+    artworks: collection.artworks?.totalCount ?? 0,
+  }
+
+  return (
+    <ArtworkFilterContextProvider
+      filters={match.location.query}
+      counts={counts}
+      sortOptions={sortOptions}
+      onChange={state => {
+        updateUrl(state, {
+          defaultValues: {
+            sort: defaultSort,
+          },
+        })
+      }}
+    >
+      <Flex
+        flexDirection="row"
+        alignItems="center"
+        justifyContent="space-between"
+      >
+        <Join separator={<Spacer x={2} />}>
+          <Text variant="lg-display">{collection.name}</Text>
+          {!collection.default && <MoreIcon width="24px" height="24px" />}
+        </Join>
+      </Flex>
+
+      <Spacer y={4} />
+
+      <SavesArtworksGridFragmentContainer
+        artworks={collection.artworks!}
+        collection={collection}
+        collectionID={collectionID}
+        relayRefetch={relay.refetch}
+      />
+    </ArtworkFilterContextProvider>
+  )
+}
+
+const QUERY = graphql`
+  query SavesArtworksQuery(
+    $collectionID: String!
+    $sort: CollectionArtworkSorts
+    $page: Int
+  ) {
+    me {
+      collection(id: $collectionID) {
+        ...SavesArtworks_collection @arguments(page: $page, sort: $sort)
+      }
+    }
+  }
+`
+
+export const SavesArtworksRefetchContainer = createRefetchContainer(
+  SavesArtworks,
+  {
+    collection: graphql`
+      fragment SavesArtworks_collection on Collection
+        @argumentDefinitions(
+          page: { type: "Int", defaultValue: 1 }
+          sort: { type: "CollectionArtworkSorts" }
+        ) {
+        name
+        default
+        artworks: artworksConnection(first: 30, page: $page, sort: $sort) {
```

Comment:
**thought (non-blocking):**  😅 still makes me uneasy to mix `first` and `page` like this, but I will leave it up to you